src_python/habitat_sim/simulator.py

```diff
@@ -738,13 +754,32 @@ def get_observation(self) -> Union[ndarray, "Tensor"]:
         return self._noise_model(obs)
 
     def _get_observation_async(self) -> Union[ndarray, "Tensor"]:
+        if self._spec.sensor_type == SensorType.AUDIO:
+            return self._get_audio_observation()
+
         if self._spec.gpu2gpu_transfer:
             obs = self._buffer.flip(0)  # type: ignore[union-attr]
         else:
             obs = np.flip(self._buffer, axis=0)
 
         return self._noise_model(obs)
 
+    def _get_audio_observation(self) -> Union[ndarray, "Tensor"]:
+        assert self._spec.sensor_type == SensorType.AUDIO
+        audio_sensor = self._agent._sensors["audio_sensor"]
+        # tell the audio sensor about the agent location
+        rot = self._agent.state.rotation
+
+        audio_sensor.setAudioListenerTransform(
+            audio_sensor.node.absolute_translation,  # set the listener position
+            np.array([rot.w, rot.x, rot.y, rot.z]),  # set the listener orientation
```

Comment:
You may want to just have the bindings take a magnum stylequaternions, that's what we do with the OBB bindings. It's a bit cleaner than having to do this cast and you can do the cast as part of a lambda with pybind11.